src/OpenTelemetry/Trace/Export/SimpleActivityProcessor.cs

```diff
@@ -0,0 +1,103 @@
+﻿// <copyright file="SimpleActivityProcessor.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+using System;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using OpenTelemetry.Internal;
+
+namespace OpenTelemetry.Trace.Export
+{
+    /// <summary>
+    /// Implements simple activity processor that exports activities in OnEnd call without batching.
+    /// </summary>
+    public class SimpleActivityProcessor : ActivityProcessor, IDisposable
+    {
+        private readonly ActivityExporter exporter;
+        private bool disposed = false;
+        private long spanCount = 0;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SimpleActivityProcessor"/> class.
+        /// </summary>
+        /// <param name="exporter">Activity exporter instance.</param>
+        public SimpleActivityProcessor(ActivityExporter exporter)
+        {
+            this.exporter = exporter ?? throw new ArgumentNullException(nameof(exporter));
+        }
+
+        public long GetSpanCount()
+        {
+            return this.spanCount;
+        }
+
+        /// <inheritdoc />
+        public override void OnStart(Activity activity)
+        {
+            Interlocked.Increment(ref this.spanCount);
+        }
+
+        /// <inheritdoc />
+        public override void OnEnd(Activity activity)
+        {
+            try
+            {
+                // do not await, just start export
```

Comment:
Curious, I think this is supposed to wait until exporter returned?

Reply:
its fire and forget, so no await.

Reply:
This seems to be different from other languages, for example [Python](https://github.com/open-telemetry/opentelemetry-python/blob/232bfdda0ccf4fc6ba6810edce87d5eb302b69d3/opentelemetry-sdk/src/opentelemetry/sdk/trace/export/__init__.py#L64).

Reply:
Without a backpressure system couldn't this become an issue if the producer overloads ability to process these `Task`s?
Perhaps at least we can keep track how how many Tasks are in-flight and drop items if it reaches a certain configurable limit?

Reply:
yes, this deserve a revisit. I followed the model from SimpleSpanProcessor.

@bruno-garcia let me know if this is a blocker for this PR or not. I can address this in next step, if you have no objections.

Reply:
Agree, without back pressure support we could easily over-produce and not understand where the blockage is. I think we need to at least have a plan for how we'd like to address it, even if the implementation comes at a later point.

Reply:
This PR is not introducing this (I am just taking current SimpleSpanProcessor implementation) - These concerns/issues exist today, and is not addressed in this PR.

We definitely need to address this, and opening an issue to track this as this shouldn't block this PR :https://github.com/open-telemetry/opentelemetry-dotnet/issues/674

Reply:
Agree that we don't want to block this PR.